### ISSUE TITLE
Crash on valid apks

### ISSUE THREAD

**Issue by thalescm** [TestArmada/flank#33]
I have a configuration like this:
```
deviceIds=Nexus6
os-version-ids=25
orientations=portrait
locales=pt_BR
shard-timeout=60
shard-duration=2400

numShards=8
shardIndex=
debug-prints=true
fetch-xml-files=false
gcloud-path=
gsutil-path=
```

and two valid apks (just ran the tests with them on my machine and all passed), but when trying to use them to shard with flank, a crash happens:
```
Exception in thread "main" java.util.EmptyStackException
	at java.util.Stack.peek(Stack.java:102)
	at java.util.Stack.pop(Stack.java:84)
	at com.walmart.otto.shards.ShardCreator.addToShard(ShardCreator.java:56)
	at com.walmart.otto.shards.ShardCreator.getShards(ShardCreator.java:39)
	at com.walmart.otto.shards.ShardExecutor.executeShards(ShardExecutor.java:43)
	at com.walmart.otto.shards.ShardExecutor.execute(ShardExecutor.java:32)
	at com.walmart.otto.Flank.main(Flank.java:48)
```

Note that the test APK has near 900 instrumentation tests (don't know if that can contribute to the problem somehow)

**Comment by bootstraponline** [comment]
What operating system are you using? If Mac, what's the output of `ulimit -a`?

**Comment by bootstraponline** [comment]
Also what version of flank are you using? Have you tried on master?

I noticed `numShards=8` so it's probably not a process limit.

**Comment by thalescm** [comment]
Using version `1.2.0`

It is Mac -> `ulimit -a`:
```
$ ulimit -a
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) 4864
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 709
virtual memory          (kbytes, -v) unlimited
```

**Comment by thalescm** [comment]
I'll clone this repo and try a run with master version

**Comment by bootstraponline** [comment]
Thanks. If it reproduces on master then I expect it's a bug in flank. 

`max user processes              (-u) 709` is kind of low. I have mine set at `2048`. However that doesn't matter for only 8 shards.

**Comment by renas** [comment]
@thalescm Would it be possible for you to share your test apk?

**Comment by thalescm** [comment]
@renas It is possible, can I send them to the email listen on your Github profile?

**Comment by thalescm** [comment]
@bootstraponline I've cloned the project but it doesn't compile due to `findBungs` hehe
I'll disable it just to check

**Comment by thalescm** [comment]
Interesting thing
I cloned the project built and ran.
Flank started to create a lot and a lot of shards, which was very weird. Then I remembered that I didn't create the `config.properties` file with the same config listed above.
  
So, when executing the default config of one test per shard, flank could do it well.
I've tried again with the same config above (8 shards) and the same crash happened.  
  
My test apk have like a lot of top packages, but only two or three of them with tests (and a lot of tests inside them). If flank does not go recursively into each package trying to find the right number of tests for each shard, this might be it.
